docs/rfcs/0020-rotate-hash-peppers

```diff
@@ -0,0 +1,365 @@
+ ##### Peering Partner (PP)
+ A Client of BFD services is commonly referred to as _peering partner_. BFD currently supports the following PPs:
+ - AB2D
+ - BCDA
+ - DPC
+ - BB2
+ ###### Symmetric Cryptography
+ Symmetric cryptography is a system of encryption where the same key is used to both encrypt and decrypt data. It is generally faster than asymmetric cryptography but is generally **considered less secure since the same key must be shared among all users**.
+ ###### Asymmetric Cryptography
+ Asymmetric cryptography is a type of encryption that uses two different keys to encrypt and decrypt data. Using a public-private keypair, anyone with the public key can send an encrypted message to the owner of the private key, who can then use the associated private key to decrypt it. The reciprocal is also true; private key holder can create an encrypted message, that holders of the public key can decrypt.
+
+## Background
+[Background]: #background
+### Current Patient Search
+[Current Patient Search]: #current_patient_search
+Prior to delving into proposed strategies or solutions, it is best to provide some context on how things currently work, and the challenges and constraints BFD currently faces.
+- BFD maintains a hashing algothim that is used to create a hashed value of a patient's MBI and/or HICN.
+- BFD shares with its peering partners all components of the hashing algorithm, including the base algorithm, a seed value (salt) and an iteration count.
+- PPs request patient information by invoking a BFD _patient search_ RESTful service (HTTP GET) that includes an obfuscated (hashed) patient identifier as a URL parameter.
+- BFD service extracts the provided patient identifier and performs one or more database lookups to derive the patients primary key identifier (BENE_ID).
+  - it first checks the _BENEFICIARIES_HISTORY_ table for a match (either MBI_HASH or BENE_CRNT_HIC_NUM); found value(s) are added to an internal list.
+  - it then checks the _BENEFICIARIES_ table for a match (either MBI_HASH or BENE_CRNT_HIC_NUM); found value(s) are added to a list. The _BENEFICIARIES_ table holds current value for a hashed MBI and a hashed HICN value.
+  - the list of hashed values+BENE_IDs is then traversed resulting in either a hit (code now has a BENE_ID) or no match (HTTP search returns NOT FOUND)
+- One thing to note, is the the use of HICN as a patient identifier is rarely (if ever) used; in fact, as of BFD v2, support for HICN lookups is technically not supported. So while this document includes handling of hashed HICN values, its inclusion in this document is perfunctory at best since no PPs actually use it.
+- The necessity for maintaining hashed MBI and HICN data represents functionality distinct to a single BFD service, _patient search_ (HTTP GET). 
```

Comment:
The PAC resources also use hashed MBI parameters.

Reply:
Yea, I sometimes forget that...did some mods to mention PACA.